File: ScalaProject/src/main/scala/com/gaia/sftm/utility/InitSpark.scala
package com.gaia.sftm.utility

import org.apache.spark.SparkConf
import org.apache.spark.sql.SparkSession

class InitSpark {

  def getSparkSession():SparkSession={

    System.setProperty("HADOOP_USER_NAME", "hadoop")
    //    下面的代码本意是想1
    //    System.setProperty("user.name", "root")

    val conf = new SparkConf
    conf.set("spark.sql.adaptive.enabled", "true")
    conf.set("spark.sql.adaptive.shuffle.targetPostShuffleInputSize", "128M")
    conf.set("spark.sql.adaptive.join.enabled", "true")
    conf.set("spark.sql.autoBroadcastJoinThreshold", "20971520")
//    conf.set("hive.metastore.uris","thrift://10.10.17.11:9083")

    val spark = SparkSession
      .builder()
      .config(conf)
//      .appName("Uni_Visit")
            .master("local[*]")
      .enableHiveSupport()
      .getOrCreate()

    spark

  }

}

File: ScalaProject/src/main/java/com/gaia/sftm/utills/InitSpark.java
package com.gaia.sftm.utills;

import org.apache.spark.sql.SparkSession;

public class InitSpark {
    public static SparkSession initSpark() {

        //  切换集群用户
        System.setProperty("HADOOP_USER_NAME", "hadoop");
        System.setProperty("user.name", "hadoop");

        //  spark 初始化
        SparkSession spark = SparkSession.builder()
                .appName("SparkToPG")
//                .master("local[*]")
                .config("spark.sql.warehouse.dir", "/warehouse/tablespace/managed/hive")
                .config("hive.metastore.uris", "thrift://192.168.0.103:9083")
                .config("spark.debug.maxToStringFields", "300")
                .config("hive.execution.engine", "mr")
                .enableHiveSupport()
                .getOrCreate();


        return spark;


    }
}
